incubator/docker-registry/templates/deployment.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Values.svcName }}
+  labels:
+    app: {{ template "name" . }} 
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.replicas }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      initContainers:
+      - name: init-myservice
+        image: "{{ .Values.initImage }}:{{ .Values.initImageVersion }}"
+        imagePullPolicy: IfNotPresent
+        command:
+          - /bin/bash
+          - -ec
+          - |
+{{- if .Values.initLoad }}
+            fn='{{ .Values.distro }}-{{ .Values.type }}-registry-{{ .Values.branch }}.tar.gz';
+            echo $fn;
+            curl {{ .Values.tarballURL }}$fn | tar -C /var/lib/registry/ -xzvf -;
+{{- end }}
+            echo Registry is all ready serving images...;
+        volumeMounts:
+        - name: image-store
+          mountPath: /var/lib/registry
+      containers:
+      - name: registry
+        image: "{{ .Values.registryImage }}:{{ .Values.registryImageVersion }}"
+        imagePullPolicy: IfNotPresent
+        resources:
+          limits:
+            cpu: 100m
+            memory: 256Mi
+        env:
+        - name: REGISTRY_HTTP_ADDR
+          value: ":4000"
+        - name: REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY
+          value: "/var/lib/registry"
+        volumeMounts:
+        - name: image-store
+          mountPath: /var/lib/registry
+        ports:
+        - containerPort: 4000
+          name: registry
+          protocol: TCP
+      volumes:
+      - name: image-store
```

Comment:
How about adding persistence options? Here's an example how this can be done:

https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-deployment.yaml#L82-L88
https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/server-pvc.yaml
https://github.com/kubernetes/charts/blob/master/stable/prometheus/values.yaml#L332-L371